### DISCUSSION TITLE
Why no Universal Mac app of the latest release?

### DISCUSSION THREAD

**Discussion by SiliconSausage** [debauchee/barrier#1380] [Q&A]
### What happened?

Just saw that there were two new releases of Barrier in quick succession; 2.3.4 and 2.4.

Since these have come a year and a half after the previous release and almost the same duration since Apple Silicon hardware has been available to the public, why aren't these releases Universal apps?

### Version

v2.3.3

### If applicable, where did you install Barrier from?

The Releases page

### What OSes are you seeing the problem on? (Check all that apply)

macOS

### What OS versions are you using?

MacOS Monterey

**Comment by shymega** [comment]
So, this would have been a better fit for the discussions section, so I've moved it here. In terms of universal mac builds, I will be looking into that, but I just haven't been able to yet - I do *not* have a mac, so I'm unable to test it myself.

**Reply by SiliconSausage** [reply]
> So, this would have been a better fit for the discussions section, so I've moved it here. 

Agreed. I'm new here and didn't know that section existed. Sorry.

**Reply by shymega** [reply]
Hey, no worries. We should add directives about the Discussions section to the README. It is on the issues template, both on the link, and at the top of the template also.

I think we'll have to rely on a macOS user to work on this. I did have a rented mac Mini, but it was too costly. Unfortunately, I couldn't use it hourly, which would have been cheaper.

**Reply by SiliconSausage** [reply]
> Hey, no worries. We should add directives about the Discussions section to the README. It is on the issues template, both on the link, and at the top of the template also.
> 
> I think we'll have to rely on a macOS user to work on this. I did have a rented mac Mini, but it was too costly. Unfortunately, I couldn't use it hourly, which would have been cheaper.

I have an M1 Mac. I'm a noob but I'd love to help. If you can point me to instructions perhaps I can help make a Universal or Apple Silicon binary available for all.

**Reply by shymega** [reply]
That would be really helpful. I think part of the problem is CI building - can you explain about the Universal Mac binary? I'd like to support some older versions of macOS, but not as far back as say, Windows XP or Vista in chronological terms.

We did have a PR recently that I merged, but I might need to revert it in light of these issues about Universal Mac binaries not being built. I'll take a look...

**Reply by shymega** [reply]
I've been working on universal builds this evening. I'm waiting for the latest iteration of my changes to build, and then @p12tic has offered to test. if you could as well, @SiliconSausage, that'd be great. I'll provide a link to the artefact once it's complete. This may not be tonight (GMT).

**Reply by shymega** [reply]
Artifact built - https://dev.azure.com/debauchee/Barrier/_build/results?buildId=649&view=artifacts&pathAsName=false&type=publishedArtifacts  - can you test it, @SiliconSausage?

**Reply by celsworth** [reply]
Hey, following this with interest as I'm proud new owner of one of the M1 Macbook Pros. So I tried this but it fails to start, and on closer inspection I'm not sure a universal binary is being built:

```
m1p:~/Downloads/Universal Binary for macOS/Barrier.app/Contents/MacOS> file barrier
barrier: Mach-O 64-bit executable x86_64
```

Compare this to a well-known mac app that is universal and we get:

```
m1p:/Applications/iTerm.app/Contents/MacOS> file iTerm2
iTerm2: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
iTerm2 (for architecture x86_64):	Mach-O 64-bit executable x86_64
iTerm2 (for architecture arm64):	Mach-O 64-bit executable arm64
```

**Reply by shymega** [reply]
Hm. I really need an M1 Mac to test this. I'll take another look tomorrow, but it seems it's not being built. I'll have to suspend the PR I made in the meantime in light of your result. Cheers!

**Comment by GoanSausage** [comment]
> Hm. I really need an M1 Mac to test this. I'll take another look tomorrow, but it seems it's not being built. I'll have to suspend the PR I made in the meantime in light of your result. Cheers!

Hi, I created this post but locked myself out of the account. The artefact you built still shows up as an Intel only binary and refuses to launch on Monterey.

**Reply by shymega** [reply]
I'm no longer involved with Barrier and have since moved to a fork. I am therefore no longer able to help and would advise waiting for someone else to help. If memory serves, the changes haven't even been put into production yet, and we need to move completely to Qt 6 for M1 to work.
